go-controller/pkg/ovn/pods.go

```diff
@@ -126,77 +126,68 @@ func (oc *Controller) syncPodsRetriable(pods []interface{}) error {
 		// is not re-added into the cache. Delete logical switch port anyway.
-		ops, err := oc.ovnNBLSPDel(logicalPort, pod.Spec.NodeName, "")
-		if err != nil {
-			klog.Errorf(err.Error())
-		} else {
-			_, err = libovsdbops.TransactAndCheck(oc.nbClient, ops)
-			if err != nil {
-				klog.Errorf("Cannot delete logical switch port %s, %v", logicalPort, err)
-			}
-		}
-
-		// Even if the port is not in the cache, IPs annotated in the Pod annotation may already be allocated,
-		// need to release them to avoid leakage.
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
-		if err == nil {
-			podIfAddrs := annotation.IPs
-			_ = oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs)
+		if err != nil {
+			return fmt.Errorf("unable to unmarshal pod annocations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
 		}
-		return
+		podIfAddrs = annotation.IPs
+	} else {
+		portUUID = portInfo.uuid
+		podIfAddrs = portInfo.ips
 	}
 
-	// FIXME: if any of these steps fails we need to stop and try again later...
 	var allOps, ops []ovsdb.Operation
-	if ops, err = oc.deletePodFromNamespace(pod.Namespace, portInfo); err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+	if ops, err = oc.deletePodFromNamespace(pod.Namespace, podIfAddrs, portUUID); err != nil {
+		return fmt.Errorf("unable to delete pod %s from namespace: %w", podDesc, err)
 	}
+	allOps = append(allOps, ops...)
 
-	ops, err = oc.ovnNBLSPDel(logicalPort, pod.Spec.NodeName, portInfo.uuid)
+	ops, err = oc.delLSPOps(logicalPort, pod.Spec.NodeName, portUUID)
 	if err != nil {
-		klog.Errorf(err.Error())
-	} else {
-		allOps = append(allOps, ops...)
+		return fmt.Errorf("failed to create delete ops for the lsp: %s: %s", logicalPort, err)
 	}
+	allOps = append(allOps, ops...)
 
 	_, err = libovsdbops.TransactAndCheck(oc.nbClient, allOps)
 	if err != nil {
-		klog.Errorf("Cannot delete logical switch port %s, %v", logicalPort, err)
+		return fmt.Errorf("cannot delete logical switch port %s, %v", logicalPort, err)
 	}
 
-	if err := oc.lsManager.ReleaseIPs(portInfo.logicalSwitch, portInfo.ips); err != nil {
-		klog.Errorf(err.Error())
+	if err := oc.lsManager.ReleaseIPs(pod.Spec.NodeName, podIfAddrs); err != nil {
+		return fmt.Errorf("cannot release IPs for pod %s: %w", podDesc, err)
 	}
 
-	if config.Gateway.DisableSNATMultipleGWs {
-		if err := deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, []*net.IPNet{}, portInfo.ips); err != nil {
-			klog.Errorf(err.Error())
+	if config.Gateway.DisableSNATMultipleGWs {			
+		if err := deletePerPodGRSNAT(oc.nbClient, pod.Spec.NodeName, []*net.IPNet{}, podIfAddrs); err != nil {
+			return fmt.Errorf("cannot delete GR SNAT for pod %s: %w", podDesc, err)
 		}
 	}
 	podNsName := ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
-	oc.deleteGWRoutesForPod(podNsName, portInfo.ips)
+	if err := oc.deleteGWRoutesForPod(podNsName, portInfo.ips); err != nil {
```

Comment:
this needs to be podIfAddrs right?

Reply:
yes omg, what was wrong with me.... too many iterations and my head wasn't in the right place or what.
thanks tim! let me do a thorough check just in case I missed anything...